syntax = "proto3";

service SpeechService {
    rpc TextToSpeech (TextRequest) returns (SpeechResponse);
    rpc SpeechToText (SpeechRequest) returns (TextResponse);
    rpc FilteredTextToSpeech (TextRequest) returns (SpeechResponse);
}

message TextRequest {
    string text = 1;
}

message SpeechRequest {
    bytes audio_data = 1;
}


message TextResponse {
    string text = 1;
}

message SpeechResponse {
    bytes audio_data = 1;
}